internal/controller/filesystemclaim_controller.go

```diff
@@ -198,16 +183,34 @@ func (r *FileSystemClaimReconciler) handleFileSystemClaimRequest(ctx context.Con
 		return ctrl.Result{Requeue: true}, nil
 	}
 
+	// Add initial overall status condition
+	fsc.Status.Conditions = utils.UpdateCondition(fsc.Status.Conditions, ConditionTypeReady, metav1.ConditionFalse, ReasonProvisioningInProgress, "Provisioning in progress", fsc.Generation)
+	if statusErr := r.Status().Update(ctx, fsc); statusErr != nil {
+		logger.Error(statusErr, "Failed to update status after initial condition")
+		return ctrl.Result{}, statusErr
+	}
+
     // Step 1: Validate nodes
     if !r.isConditionTrue(fsc, ConditionTypeNodeValidated) {
 	// Validate node exists in the cluster
 		err := r.validateNode(ctx, fsc)
 		if err != nil {
 			logger.Error(err, "Node validation failed")
 			fsc.Status.Conditions = utils.UpdateCondition(fsc.Status.Conditions, ConditionTypeNodeValidated, metav1.ConditionFalse, ReasonNodeValidationFailed, err.Error(), fsc.Generation)
+			fsc.Status.Conditions = utils.UpdateCondition(fsc.Status.Conditions, ConditionTypeReady, metav1.ConditionFalse, ReasonProvisioningFailed, "Node validation failed", fsc.Generation)
+			if statusErr := r.Status().Update(ctx, fsc); statusErr != nil {
```

Comment:
As above.  Only update if something changed, and exit after attempting an update if it did. The only thing that changes is the result+error combination that you return